core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/UniqueIDReadOnlyLDAPUserStoreManager.java

```diff
@@ -1446,22 +1464,57 @@ public boolean isReadOnly() throws UserStoreException {
         return values;
     }
 
+    /**
+     * This method supports multi-attribute filters with paginated search for user(s).
+     *
+     * @param condition   Validated Condition tree.
+     * @param profileName Default profile name.
+     * @param limit       The number of entries to return in a page.
+     * @param offset      Start index.
+     * @param sortBy      Sort according to the given attribute name.
+     * @param sortOrder   Sorting order.
+     * @return A non-null PaginatedSearchResult instance. Typically contains user names with pagination.
+     * @throws UserStoreException If an UserStoreException is encountered
+     *                            while searching for users in a given condition.
+     */
     @Override
     protected UniqueIDPaginatedSearchResult doGetUserListWithID(Condition condition, String profileName, int limit,
             int offset, String sortBy, String sortOrder) throws UserStoreException {
 
-        // TODO: Need to improve this method to get the userID as well.
-        PaginatedSearchResult userNames = super.doGetUserList(condition, profileName, limit, offset, sortBy, sortOrder);
-        UniqueIDPaginatedSearchResult userList = new UniqueIDPaginatedSearchResult();
-        userList.setPaginatedSearchResult(userNames);
-        userList.setSkippedUserCount(userNames.getSkippedUserCount());
-        List<User> users = new ArrayList<>();
-        for (String userName : userNames.getUsers()) {
-            User user = getUser(null, userName);
-            users.add(user);
+        UniqueIDPaginatedSearchResult result = new UniqueIDPaginatedSearchResult();
```

Comment:
Need to format

Reply:
Addressed in https://github.com/wso2/carbon-kernel/pull/2866/commits/c80f2b70a66acc006d2ed0d1fa5b20dfc04bb81d